### PR TITLE
Seed keyword passed

### PR DESCRIPTION
**Pull request by George3d6**
Batch improvements based on suggestions from @hakunanatasha after her extensive review of the codebase.

* Seed function no longer has a default value for the seed, the seed_nr being passed is configurable from all of the functions that `seed` is called in.

### CODE REVIEW

File: lightwood/api/json_ai.py [hunk 1]
```diff
@@ -627,7 +627,7 @@ class Predictor(PredictorInterface):
     mode: str
 
     def __init__(self):
-        seed()
+        seed({json_ai.problem_definition.seed_nr})
```
Comment (hakunanatasha) [inline lightwood/api/json_ai.py:630]: is json_ai global? How is this inherited? Shouldn't it be set from self?

File: lightwood/helpers/seed.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import random
 
 
-def seed(seed_nr=420):
+def seed(seed_nr):
```
Comment (hakunanatasha) [inline lightwood/helpers/seed.py:6]: type hint this

**Review by hakunanatasha — CHANGES_REQUESTED** [review]
Mostly good - only issue is with json_ai; should this not be set as self.seed? If not, should be fine - will approve after comments addressed.

File: lightwood/api/json_ai.py (hunk 1 above)
Comment (George3d6) [reply lightwood/api/json_ai.py:630]: No... this is happening in `json_ai.py` when we are constructing the code for the Predictor class.

Values specified in the problem_definition get inlined in the Predictor class.

I'm not sure the currently built predictor classes are the most readable... but the current questions we are focusing on are more towards json ai design, so I'd rather tackle the generated predictors afterwards.

Comment (hakunanatasha) [reply lightwood/api/json_ai.py:630]: got it

**Review by hakunanatasha — APPROVED** [review]
This is fine to merge.

**Pull request merged**